pkg/probe/http/http_test.go

```diff
@@ -314,3 +314,51 @@ func TestHTTPProbeChecker_NonLocalRedirects(t *testing.T) {
 		})
 	}
 }
+
+func TestHTTPProbeChecker_HostHeaderPreservedAfterRedirect(t *testing.T) {
+	successHostHeader := "www.success.com"
+	failHostHeader := "www.fail.com"
+
+	handler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		switch r.URL.Path {
+		case "/redirect":
+			http.Redirect(w, r, "/success", http.StatusFound)
+		case "/success":
+			if r.Host == successHostHeader {
+				w.WriteHeader(http.StatusOK)
+			} else {
+				http.Error(w, "", http.StatusBadRequest)
+			}
+		default:
+			http.Error(w, "", http.StatusInternalServerError)
+		}
+	})
+	server := httptest.NewServer(handler)
+	defer server.Close()
+
+	testCases := map[string]struct {
+		hostHeader     string
+		expectedResult probe.Result
+	}{
+		"success": {successHostHeader, probe.Success},
+		"fail":    {failHostHeader, probe.Failure},
+	}
+	for desc, test := range testCases {
+		headers := http.Header{}
+		headers.Add("Host", test.hostHeader)
+		t.Run(desc+"local", func(t *testing.T) {
+			prober := New(false)
```

Comment:
Would you mind naming these variables (i.e. `followNonLocalRedirects = false`)?

I created https://github.com/kubernetes/kubernetes/pull/78031/files to make a similar change for other tests in this repo.

Reply:
Done.